Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -426,8 +434,30 @@ void FieldProbe::ComputeDiags (int step)
         for (MyParIter pti(m_probe, lev); pti.isValid(); ++pti)
         {
             const auto getPosition = GetParticlePosition(pti);
-            auto const np = pti.numParticles();
+                auto setPosition = SetParticlePosition(pti);
 
+            auto const np = pti.numParticles();
+            if (do_moving_window_FP && step > warpx.start_moving_window_step && step <= warpx.end_moving_window_step)
+            {
+                int step_diff = step - m_last_compute_step;
+                for (auto ip=0; ip < np; ip++)
```

Comment:
You can replace this serial for with an `amrex::ParallelFor` :)

That way, the function operates directly on GPU (when available) and is also faster.
```suggestion
                for::ParallelFor( np, [=] AMREX_GPU_DEVICE (long ip)
```